# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/JitActivation.cpp
# Commit: 40031ad88dbd
# Full Hash: 40031ad88dbd9af3296493ae4bf501c158cb0ca2
# Author: Dmitry Bezhetskov <dbezhetskov@igalia.com>
# Date: 2020-11-03 21:39:33
# Regressor Bug: 1639153
# File Overlap Count: 1
# Description:
#   Bug 1639153 - Part 5: Remove remaining uses of Frame::tls. r=lth
#   
#   Here we remove remaining uses of Frame::tls. There are many places where
#   we use it: in debug frames, in profiling frames, in jit activation, etc.
#   All these places require short fixes to use our new scheme for getting
# ==============================================================================

diff -r cee7eaa8c402 -r 40031ad88dbd js/src/vm/JitActivation.cpp
--- a/js/src/vm/JitActivation.cpp	Tue Nov 03 10:25:14 2020 +0000
+++ b/js/src/vm/JitActivation.cpp	Tue Nov 03 10:28:10 2020 +0000
@@ -233,7 +233,7 @@
   void* pc = unwindState.pc;
   const wasm::Frame* fp = wasm::Frame::fromUntaggedWasmExitFP(unwindState.fp);
 
-  const wasm::Code& code = fp->instance()->code();
+  const wasm::Code& code = wasm::GetNearestEffectiveTls(fp)->instance->code();
   MOZ_RELEASE_ASSERT(&code == wasm::LookupCode(pc));
 
   // If the frame was unwound, the bytecodeOffset must be recovered from the